### PR TITLE
Feat: Ignore Columns

### PR DESCRIPTION
**Pull request by Codexiaoyi**
ignore columns by Tag and Option.

### CODE REVIEW

**Comment by Codexiaoyi** [comment]
#47

**Comment by flycash** [comment]
Please fix deepsource issues

File: model.go [hunk 1]
```diff
@@ -72,6 +72,11 @@ func (t *tagMetaRegistry) Register(table interface{}, opts ...tableMetaOption) (
 	for i := 0; i < lens; i++ {
 		structField := v.Field(i)
 		tag := structField.Tag.Get("eql")
+		isIgnore := strings.Contains(tag, "-")
```
Comment (flycash) [inline model.go:75]: please help to refactor this part. I think using `strings.Contains` is not a good practice

File: model.go [hunk 2]
```diff
@@ -97,3 +102,28 @@ func (t *tagMetaRegistry) Register(table interface{}, opts ...tableMetaOption) (
...
 }
+
+// IgnoreFieldsOption function provide an option to ignore some fields when register table.
+// Ignore fields by field name when the field exist in TableMeta.
+func IgnoreFieldsOption(fieldNames ...string) tableMetaOption {
+	return func(meta *TableMeta) {
+		for _, field := range fieldNames {
+			//has field in the TableMeta
+			if _, ok := meta.fieldMap[field]; ok {
+				exist := false
```
Comment (flycash) [inline model.go:113]: I think we don't need this flag, just always `delete(meta.fieldMap, field). And you need to check if this is primary key, I think we'd better not allow users to remove primary key.

Comment (Codexiaoyi) [reply model.go:113]: ok.

File: model.go [hunk 3]
```diff
@@ -95,5 +106,46 @@ func (t *tagMetaRegistry) Register(table interface{}, opts ...tableMetaOption) (
 	}
 	t.metas.Store(rtype, tableMeta)
 	return tableMeta, nil
+}
 
+// The tag is parsed by bit operation.
+// This is better than using strings.Contains queries once for each tag.
+func parseTag(tag string) tagBitFlags {
```
Comment (flycash) [inline model.go:113]: hhh, you don't need this. Just iterate the tags and the modify the column meta directly

File: model_test.go [hunk 4]
```diff
@@ -56,3 +56,54 @@ func TestTagMetaRegistry(t *testing.T) {
...
+
+	_, optionExist := meta.fieldMap["FirstName"]
+	assert.False(t, optionExist)
+
+	_, tagExist := meta.fieldMap["Age"]
+	assert.False(t, tagExist)
+}
+
+type TestIgnoreModel struct {
+	Id        int64 `eql:"auto_increment,primary_key,-"`
```
Comment (flycash) [inline model_test.go:105]: I prefer that symbol `-` has highest priority. So if we find `-`, we just ignore it.

File: model.go [hunk 5]
```diff
@@ -95,5 +106,46 @@ func (t *tagMetaRegistry) Register(table interface{}, opts ...tableMetaOption) (
...
+}
+
+// IgnoreFieldsOption function provide an option to ignore some fields when register table.
+// Ignore fields by field name when the field is not primary key.
+func IgnoreFieldsOption(fieldNames ...string) TableMetaOption {
+	return func(meta *TableMeta) {
+		for _, field := range fieldNames {
+			//has field in the TableMeta
+			if columnMeta, ok := meta.fieldMap[field]; ok {
+				if columnMeta.isPrimaryKey {
```
Comment (flycash) [inline model.go:135]: don't need to check this. Users should be responsible for their action. They can find they remove the PK when they run tests.

**Review by flycash — COMMENTED** [review]
Sorry for late reply because I was busy this month

File: model.go (hunk 5 above)
Comment (Codexiaoyi) [reply model.go:135]: yes.

**Review by flycash — APPROVED** [review]

**Comment by flycash** [comment]
Please add example for this. I have added examples for some public functions and you can take them as reference

**Comment by Codexiaoyi** [comment]
> Please add example for this. I have added examples for some public functions and you can take them as reference

ok

**Pull request merged**